src/debuggability.js

```diff
@@ -141,6 +141,63 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+var supportedEvents = ["created", "chained", "fulfilled",
+    "rejected", "resolved", "cancelled"];
+
+Promise.on = function (eventName, hookFunction, simpleEventApi) {
+    if (supportedEvents.indexOf(eventName) < 0) {
+        throw new Error("You can only subscribe to these events:" +
+            supportedEvents);
+    }
+    var newHook = null;
+    if (simpleEventApi) {
+        // This interface allows users to subscribe to promises
+        // related events in more performable way then event API
+        // hookFunction is called after the event it has this
+        // (current promise reference) defined and single
+        // argument (child promise in "chained" event)
+        newHook = hookFunction;
+    } else {
+        newHook = function (other) {
+            hookFunction({
+                guid: this._promiseId ? this._promiseId : null,
+                childGuid: (other && other._promiseId) ?
+                    other._promiseId : null,
+                eventName: eventName,
+                detail: this.isRejected() ? this.reason :
+                    this.isFulfilled() ? this.value : null,
+                //event.label //is this relevant?
+                timeStamp: Date.now(),
+                stack: this._trace ? this._trace : null
+            });
+        };
+        hookFunction._eventHandler = newHook;
+    }
+    util.hookTo(Promise.prototype, "_hook_" + eventName, newHook);
+};
+
+Promise.off = function (eventName, hookFunction) {
+    if (supportedEvents.indexOf(eventName) < 0) {
+        throw new Error("You can only subscribe to these events:"
+            + supportedEvents);
+    }
+    util.unhookFrom(Promise.prototype, "_hook_" + eventName,
+        hookFunction._eventHandler ? hookFunction._eventHandler :
+            hookFunction);
+};
+
+// Event handlers can be placed on (window|process).__PROMISE_INSTRUMENTATION__
+// object
+if (util.global && typeof util.global.__PROMISE_INSTRUMENTATION__ ===
+    "object") {
+    var callbacks = util.global.__PROMISE_INSTRUMENTATION__;
+    for (var eventName in callbacks) {
```

Comment:
If you mean __PROMISE_INSTRUMENTATION__ object than I will document it as soon as we decide to keep going with this (__PROMISE_INSTRUMENTATION__ on window) option (see previous comment).